api/client/gro_client.py

```diff
@@ -877,8 +877,32 @@ def get_df(self, show_revisions=False, index_by_series=False):
         if index_by_series and not self._data_frame.empty:
             columns = intersect(DATA_SERIES_UNIQUE_TYPES_ID, self._data_frame.columns)
             if len(columns) > 0:
-                indexed_df = self._data_frame.set_index(columns)
+                # add entity names to the dataframe
+                series_df = pandas.DataFrame([dict(s) for s in self.get_data_series_list()])
+                name_columns = []
+                for col in columns:
+                    name_col = col.replace('id', 'name')
+                    name_columns.append(name_col)
+                    # lookup names if not available in the series list (e.g. if input is GDH)
+                    if name_col not in series_df.columns or series_df[name_col].isna().any():
+                        entity_lookup = {
+                            entity_id: self.lookup(ENTITY_KEY_TO_TYPE[col], entity_id)['name']
+                            for entity_id in list(series_df[col].unique())
+                        }
+                        series_df[name_col] = series_df[col].apply(lambda x: entity_lookup[x])
+
+                df_with_names = self._data_frame.merge(series_df[columns + name_columns],
+                                                       on=columns, copy=False)
+
+                indexed_df = df_with_names.set_index(columns)
                 indexed_df.index.set_names(DATA_SERIES_UNIQUE_TYPES_ID, inplace=True)
+
+                # move entity names to column headers
+                indexed_cols = pandas.MultiIndex.from_frame(indexed_df[name_columns])
+                indexed_df.drop(name_columns, axis=1, inplace=True)
+
+                # indexed_df = pandas.concat([indexed_df], keys=indexed_cols, axis=1)
+                # indexed_df.columns.names = name_columns + ['entities']
```

Comment:
I commented out these two lines because they were causing tests to fail. Without them, it seems to work, but I'm still trying to grok what the intent here is

Reply:
that's where the entity names are added to the column headers. removing these lines defeats the purpose of this PR, or we need to find a different way to accomplish it